server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -356,7 +354,12 @@ private void waitDecommissionFinish() {
     while (isDecommissioning()) {
       remainApplicationNum = shuffleTaskManager.getAppIds().size();
       if (remainApplicationNum == 0) {
-        serverStatus.set(ServerStatus.DECOMMISSIONED);
+        boolean wasDecommissioning =
+            serverStatus.compareAndSet(ServerStatus.DECOMMISSIONING, ServerStatus.DECOMMISSIONED);
+        if (!wasDecommissioning) {
+          break;
```

Comment:
I think we should print some logs here.

Reply:
We are leaving the lop here and there will be a log when exiting this method:

    LOG.info("Decommission exiting, remaining {} applications not finished.", remainApplicationNum);

Reply:
`remainApplicationNum` is 0, so this line will not be reached. right?

Reply:
What about
```suggestion
          LOG.info("Ready to decommission but decommissioning state left unexpectedly.");
          break;
```

Reply:
> What about

It is ok for me.

Reply:
You are right, we won't see that existing log message.

I have moved the `"All applications finished. Current status is " + serverStatus` message up so it is printed in any case logging the current stats, and added the suggested additional log line.